### PR TITLE
Add `rs.wordpress.api:android` dependency

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR adds the WordPress.rs library and uses it to attempt URL auto discovery for new XMLRPC sites that are added to the apps. This should be a very small number of users, and it's performed in the background so it shouldn't affect UX.

To Test:
1. Run this PR on a device or emulator
2. Add a new XMLRPC site (vanilla.wpmt.co is a good one)
3. Note in logcat that the auto discovery is successful
4. Try a site that doesn't support auto discovery (google.com is a good one)
5. Note in logcat that the auto discovery is not successful

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/289">25.3</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: settings.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 pluginManagement {
-    gradle.ext.kotlinVersion = '1.9.22'
-    gradle.ext.kspVersion = '1.9.22-1.0.17'
+    gradle.ext.kotlinVersion = '1.9.24'
```
Comment (nbradbury) [inline settings.gradle:2]: Do we really want to update the `kotlinVersion` and `kspVersion` in this PR? That seems like something that should be handled separately.

Likewise with `androidxComposeCompilerVersion`.

**Review by nbradbury — CHANGES_REQUESTED** [review]
This works as expected when I use `vanilla.wpmt.co` but when I use `google.com`, instead of seeing "auto discovery is not successful" in logcat, I see "onDiscoveryResponse has error: XMLRPC_BLOCKED - XMLRPC_BLOCKED." 

In fact, in this situation `LoginActivity.gotXmlRpcEndpoint` isn't called. Is this expected?

File: settings.gradle (hunk 1 above)
Comment (jkmassel) [reply settings.gradle:2]: This was bumped to align with the https://github.com/automattic/wordpress-rs project,t which used very slightly newer versions

**Comment by jkmassel** [comment]
Ah, yep - this is expected, sorry - `LoginActivity.gotXmlRpcEndpoint` is only called if the endpoint is valid – I should've noted that any other output is indicative of a failure.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/296">25.6</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by nbradbury — APPROVED** [review]
@jkmassel This is looking good to go! Feel free to merge when ready. :shipit:

**Pull request merged**